content/docs/latest/setup/listeners/mtls.md

```diff
@@ -0,0 +1,456 @@
+   ```sh
+   export TLS_CERT="$(< example_certs/gateway.crt)"  
+   export TLS_KEY="$(< example_certs/gateway.key)"  
+   export CA_CERT="$(< example_certs/example.com.crt)"
+   
+   cat <<EOF | kubectl apply -f -
+   apiVersion: v1
+   kind: Secret
+   metadata:
+     name: https
+     namespace: {{< reuse "docs/snippets/namespace.md" >}}
+     labels: 
+       gateway: https
+   type: kubernetes.io/tls
+   stringData:
+     tls.crt: |
+   $(echo "$TLS_CERT" | sed 's/^/    /')
+     tls.key: |
+   $(echo "$TLS_KEY" | sed 's/^/    /')
+     ca.crt: | 
+   $(echo "$CA_CERT" | sed 's/^/    /')
+   EOF
+   ```
+
+4. Create a client certificate and private key. You use these credentials later when sending a request to the gateway proxy. The client certificate is signed with the same root CA certificate that you used for the gateway proxy. 
+   ```sh
+   openssl req -out example_certs/client.example.com.csr -newkey rsa:2048 -nodes -keyout example_certs/client.example.com.key -subj "/CN=client.example.com/O=client organization"
+   openssl x509 -req -sha256 -days 365 -CA example_certs/example.com.crt -CAkey example_certs/example.com.key -set_serial 1 -in example_certs/client.example.com.csr -out example_certs/client.example.com.crt
+   ```
+   
+## Set up an mTLS listener
+
+1. Create a mTLS Gateway that is configured with the TLS certificates that you set up earlier. To use the default Envoy-based kgateway proxy, set the gatewayClassName to `{{< reuse "docs/snippets/gatewayclass.md" >}}`. To use agentgateway, set the gatewayClassName to `{{< reuse "docs/snippets/agw-gatewayclass.md" >}}`.
+
+   {{< tabs tabTotal="2" items="Gateway listeners,ListenerSet (experimental)">}}
+   {{% tab tabName="Gateway listeners" %}}
+
+   ```yaml
+   kubectl apply -f- <<EOF
+   apiVersion: gateway.networking.k8s.io/v1
+   kind: Gateway
+   metadata:
+     name: https
+     namespace: {{< reuse "docs/snippets/namespace.md" >}}
+     labels:
+       gateway: https
+   spec:
+     gatewayClassName: {{< reuse "docs/snippets/gatewayclass.md" >}}
+     listeners:
+       - name: https
+         port: 443
+         protocol: HTTPS
+         hostname: https.example.com
+         tls:
+           mode: Terminate
+           certificateRefs:
+             - name: https
+               kind: Secret
+         allowedRoutes:
+           namespaces:
+             from: All
+   EOF
+   ```
+
+   |Setting|Description|
+   |--|--|
+   |`spec.gatewayClassName`|The name of the Kubernetes gateway class that you want to use to configure the gateway. |
+   |`spec.listeners`|Configure the listeners for this gateway. In this example, you configure an HTTPS gateway that listens for incoming traffic on port 443. |
+   |`spec.listeners.tls.mode`|The TLS mode that you want to use for incoming requests. In this example, HTTPS requests are terminated at the gateway and the unencrypted request is forwarded to the service in the cluster. |
```

Comment:
Is it worthwhile to say the other modes that are supported? or I guess it's just passthrough per this API ref doc: https://gateway-api.sigs.k8s.io/reference/1.4/spec/#listenertlsconfig